### PR TITLE
Adding Browserify to the build process

### PR DESCRIPTION
**Pull request by typerqeo**
- Split up the set of JS files into two groups: refactored files and global (legacy) files.

- Changed the build process to be the following:

    1. delete dist/
    1. lint the refactored JS files
    1. concatenate and lint the legacy JS files, and output to dist/gen/index.js
    1. copy the refactored files to dist/gen/
    1. bundle the JS files together using Browserify, with the generated index.js as the entry point
    1. copy html and static files to dist/
    1. compile and copy sass to dist/css

- Made db.js into an ES6 module. I chose to start with this file since it's already pretty small and self-contained. The global `dbSnapshot` was replaced with getter and setter functions. Eventually all of the JS files should be made into ES6 modules.

- Added global-dependencies.js, which imports all the dependencies that the legacy files use. These dependencies are both external (from npm) and internal (from refactored JS files).

- Added exports.js to expose any debug info you'd want to have in the browser console. In this PR, we're exporting a `snapshot()` function which exposes the `dbSnapshot`.

- Added Babelify to use ES6 syntax with Browserify. This adds more lines to the generated JS, which is not good, but it's possible to reduce the size eventually.

This PR is part of the refactoring effort in https://github.com/Miodec/monkeytype/issues/495

### CODE REVIEW

**Comment by Marco-Pagani** [comment]
you mentioned in #495 using webpack, is building with browserify an intermediate step so that you can keep the gulp file for now? Or do you plan to stick with this instead of webpack?

**Comment by typerqeo** [comment]
Yeah, I'd like us to use Browserify at least until we can get rid of Gulp (specifically, the JS concatenation part). Then we can switch to Webpack more easily.

File: src/js/account.js [hunk 1]
```diff
@@ -284,7 +284,7 @@ firebase.auth().onAuthStateChanged(function (user) {
       verifyUser(verifyUserWhenLoggedIn).then((data) => {
         showNotification(data.data.message, 3000);
         if (data.data.status === 1) {
-          dbSnapshot.discordId = data.data.did;
+          db_getSnapshot().discordId = data.data.did;
```
Comment (Miodec) [inline src/js/account.js:287]: Will this make a reference to the object or a copy? Im thinking if setting `discordId` will work like this.

Comment (typerqeo) [reply src/js/account.js:287]: `db_getSnapshot()` should return the reference (by value), so setting a property of the snapshot should work. It's similar to

```
const snapshot = db_getSnapshot();
snapshot.discordId = data.data.did;
```

Note however that assigning the snapshot from the getter (`db_getSnapshot() = otherSnapshot;`) won't work. That's why we also have `db_setSnapshot(otherSnapshot)`.

**Pull request merged**